### PR TITLE
Added support for show functions

### LINKED ISSUE #186: Support show functions in design document module
Supporting show functions is part of a larger work item for design document sub-objects (#137).
- In database module, query _show endpoint: i.e. `get_show_result(show_name, ddoc_show_id, doc_id,  **kwargs)`
- In design document module, support show functions with a `shows` property accessor and management functions

### PR DESCRIPTION
**Pull request by emlaver**
## What

Add support for managing and querying show functions.

## How

- Handled show function execution against the _show endpoint in `CouchDatabase.get_show_function_result`
- Added support for managing show functions in design document

## Testing

Unit test cases to assert show function management in DesignDocumentTests.
Unit test cases to assert `get_show_function_result` in DatabaseTests.

## Issues
fixes #186

### CODE REVIEW

**Comment by rhyshort** [comment]
+1

File: src/cloudant/design_document.py [hunk 1]
```diff
@@ -522,6 +595,17 @@ def get_list_function(self, list_name):
...
         return self.lists.get(list_name)
 
+    def get_show_function(self, show_name):
+        """
+        Retrieves a specific show function from the locally cached DesignDocument
+        shows dictionary by name.
+
+        :param str show_name: Name used to identify the show function.
+
+        :returns: Index dictionary for the specified show function name
```
Comment (ricellis) [inline src/cloudant/design_document.py]: I'm a bit confused by `Index dictionary` in this returns statement. Doesn't it just return the string of the function?
I note that both `get_index`(self, index_name) and `get_list_function(self, list_name)` use the same terminology, but I think it only makes sense in the index case so probably need to fix this and the `list` case.

**Comment by ricellis** [comment]
+1 besides that one doc thing I noted.

File: src/cloudant/design_document.py (hunk 1 above)
Comment (emlaver) [inline src/cloudant/design_document.py]: Updated in https://github.com/cloudant/python-cloudant/pull/196/commits/22c15e4a2de6c80755c6e1aa61fb6c44f95b9c83#diff-f2994c8b7af6529726f4b4fbdd48e119R594 and https://github.com/cloudant/python-cloudant/pull/196/commits/22c15e4a2de6c80755c6e1aa61fb6c44f95b9c83#diff-f2994c8b7af6529726f4b4fbdd48e119R594

**Pull request merged**